### ISSUE TITLE
Installations and updates are executed despite MSFS open

### ISSUE THREAD

**Issue by FoxtrotSierra6829** [flybywiresim/installer#463]
### Installer Version

v3.4.0

### Describe the bug

Updating MSFS addons is not being prevented when MSFS is open, which can lead to file corruption and errors in the VFS.

### Expected behavior

All forms of installation and uninstallation conflicting with an external application should be unavailable while the external application is open.

### Steps to reproduce

- have MSFS open
- update e.g. A32NX
- update is successfully executed

### References (optional)

![grafik](https://github.com/flybywiresim/installer/assets/50082450/9d738d53-488b-445f-b4c7-a802f4a6184c)
![grafik](https://github.com/flybywiresim/installer/assets/50082450/aa0f73d3-8ac1-4672-9e47-2d80bc42a087)

### Additional info (optional)

Suspect functionality was compromised in a past refactor.

### Discord Username (optional)

foxtrotsierra

**Comment by tracernz** [comment]
Would be good to have an “are you sure?” red warning rather than outright refusing.

**Comment by FoxtrotSierra6829** [comment]
> Would be good to have an “are you sure?” red warning rather than outright refusing.

Tbh, updating and uninstalling should be outright refused while MSFS is open. Installing, okay, it's not gonna destroy anything, the addon is just not going to appear in the current instance.

**Comment by tracernz** [comment]
Restarting a flight will reload most things from disk, and quick reload in the behaviour window reloads everything else from disk, so it’s quite useful for power users to be able to do it.

**Comment by FoxtrotSierra6829** [comment]
We're frequently discussing with users if removing all other addons except for the A32NX means that there can still be other aircraft, sceneries and liveries in the community folder or not. I don't think we can expect everyone to reload via the dev tools after updating.

**Comment by tracernz** [comment]
Power users are not the ones coming for support.

**Comment by FoxtrotSierra6829** [comment]
> Power users are not the ones coming for support.

Yeah but we don't only build the installer for power users. In that regard, protecting the average end user and lowering the support volume is of higher priority to me. Maybe we can think about an override switch in the developer settings.

**Comment by tracernz** [comment]
MSFS addon linker does exactly what I had in mind, only we could be a little more forceful with colour.
![image](https://github.com/flybywiresim/installer/assets/9995998/b3f40662-d2dc-4598-ad74-fb60c0874ac2)

**Comment by alepouna** [comment]
Both points are valid to be honest. The poweruser will know the risks, but the average Joe might just skip the warning because 'big red button i must press!' which would not be handy for support, though I'm sure by year 4 of MSFS even new users should know you need to restart for things to show up from your community folder. 

A timer or a require tickbox (everytime) before you can continue, or a setting in an advanced section of the installer would work both ways I think. Though to be fair, the "powerusers" might build from source, or download directly from GitHub rather than use our installer.
